src/components/InputProblem.tsx

```diff
@@ -105,11 +122,56 @@ export const InputProblem = ({
     return trimmedInput.toLowerCase() === trimmedAnswer.toLowerCase()
   }
 
+  const handleFeedback = (userResponse: string, userAttempts: number): string => {
+    if (evaluateInput(userResponse, solution)) {
+      setFeedback(correctResponse)
+    } else if (retryLimit === 0 || userAttempts !== retryLimit + 1) {
```

Comment:
Should this not be `(retryLimit === 0 || retriesAllowed !== retryLimit)`? If so, does `userAttempts` even need to get passed here?

Reply:
Oh -- wait, I think you have to pass it for the `getPersistedState` path invocation. I'm still not uncertain there's not a potential off-by-one error being introduced, and it makes me nervous that it's not obvious either way :sweat_smile: .

Reply:
That's right. If I were to do `retriesAllowed !== retryLimit` then when refreshing on the attempt before the last attempt you will get the `attemptsExhaustedResponse` even though the user should have one more attempt. Would we want to forego displaying the feedback on refresh? Then the user would only see the feedback after they submit.

Reply:
Ah, what about if in `handleSubmit()` we always invoke `handleFeedback(values.response, retriesAllowed)`. In fact, in that case we can just pull it out and do it with `setResponse`:

```javascript
    handleFeedback(values.response, retriesAllowed)
    setResponse(values.response)
```

...then `handleFeedback` can use the same conditional logic:

```javascript
  const handleFeedback = (userResponse: string, userAttempts: number): void => {
    if (evaluateInput(userResponse, solution)) {
      setFeedback(correctResponse)
    } else if (retryLimit === 0 || userAttempts !== retryLimit) {
      setFeedback(determineFeedback(userResponse, encourageResponse, answerResponses, evaluateInput))
    } else {
      setFeedback(attemptsExhaustedResponse)
    }
  }
```

The wrinkle is really on the `getPersistedState` path, but I think we can handle that by invoking the method slightly differently there:

```javascript
handleFeedback(parsedPersistedState.userResponse, parsedPersistedState.retriesAllowed > 0 ? parsedPersistedState.retriesAllowed - 1 : 0)
```

The idea is that, in essence, on the rehydration path to invoke the `handleFeedback` the way it was originally, we need to "rewind by one". The ternary checks for 0 as I suspect that's what we'll reset to eventually when we implement [k12-493](https://github.com/openstax/k12/issues/493). 

Does that work, or is there some bug it now introduces? :thinking:

Reply:
It appears to work for `InputProblem.tsx`. I'll update the logic accordingly across the other components (except for `UserInputBlock.tsx`).

I understand how this refactor essentially uses the same version of `retriesAllowed` without the persistor, but how is adding 1 to `retryLimit` paired with the updated `retriesAllowed` different from subtracting 1 from `retriesAllowed`? Is this the preferred method because we know functionally it should operate similarly to the non-persistor version of the component?

Reply:
I think it's just confusing to have:

1) (Needlessly) different invocations to `handleFeedback()` in `handleSubmit()`. It makes a reader think something special is happening, but there's really not. In general, it seems much cleaner (subjective) to let the `getPersistedState` do "something special" vs muddy up the `handleSubmit()` path. Said differently, it's overly confusing to do a '+1' and then (the algebraic equivalent of) '-1' in the same code path. 
2) Since, in this incarnation, we're reproducing the same `if / else / ...` statement in `handleSubmit()` and `handleFeedback()`, it's very confusing to a reader that there is a different conditional statement IMHO. My concern is it could lead to "off by one" errors, and it definitely wasn't clear to me when reviewing that your change hadn't introduced one.

Reply:
Thanks for the explanation! I see how my implementation provides less clarity and could lead the reader down an unintended path.